acmens.py

```diff
@@ -200,42 +202,56 @@ def _poll_until_not(url, pending_statuses, err_msg):
         order["authorizations"][0], None, "Error getting challenges"
     )
 
-    challenge = [c for c in chl_result["challenges"] if c["type"] == "http-01"][0]
+    type_id = 'dns-01' if challenge_type == 'dns' else 'http-01'
+    challenge = [c for c in chl_result["challenges"] if c["type"] == type_id][0]
     token = re.sub(r"[^A-Za-z0-9_\-]", "_", challenge["token"])
     keyauthorization = "{0}.{1}".format(challenge["token"], thumbprint)
 
     # build request for the server to test this challenge.
     test_url = challenge["url"]
     test_raw = "{}"
 
-    # challenge response for server
-    response = {
-        "uri": ".well-known/acme-challenge/{0}".format(challenge["token"]),
-        "data": keyauthorization,
-    }
-
     # Step 6: Ask the user to host the token on their server
-    sys.stderr.write(
+    if challenge_type == 'dns':
+        sys.stderr.write(
+        """\
+Please update your DNS to have the following TXT record:
+
+--------------
+DNS TXT record: _acme-challenge.{domain}
+DNS TXT record contents: \"{token}\"
+--------------
+
+Notes:
+- Do not include the quotes in the TXT record.
+
+""".format(domain=domain.replace('*.', ''), token=challenge["token"]))
```

Comment:
The value of the DNS TXT record must be  [`base64(sha256(key_authorization))`][1]

You may accomplish this with something like:

```python
_b64(hashlib.sha256(keyauthorization.encode()).digest())
```

[1]: https://tools.ietf.org/html/rfc8555#section-8.4

Reply:
Also since the DNS TXT record value is not a token but base64 encoded sha256 digest of the key authorization, can we rename the `token` arg to `keyauth`?